aws/resource_aws_sagemaker_endpoint_configuration.go

```diff
@@ -0,0 +1,278 @@
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+						},
+					},
+				},
+			},
+
+			"kms_key_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceAwsSagemakerEndpointConfigurationCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).sagemakerconn
+
+	var name string
+	if v, ok := d.GetOk("name"); ok {
+		name = v.(string)
+	} else {
+		name = resource.UniqueId()
+	}
+
+	createOpts := &sagemaker.CreateEndpointConfigInput{
+		EndpointConfigName: aws.String(name),
+	}
+
+	prodVariants, err := expandProductionVariants(d.Get("production_variants").([]interface{}))
+	if err != nil {
+		return err
+	}
+	createOpts.ProductionVariants = prodVariants
+
+	if v, ok := d.GetOk("kms_key_id"); ok {
+		createOpts.KmsKeyId = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("tags"); ok {
+		createOpts.Tags = tagsFromMapSagemaker(v.(map[string]interface{}))
+	}
+
+	log.Printf("[DEBUG] Sagemaker endpoint configuration create config: %#v", *createOpts)
+	_, err = conn.CreateEndpointConfig(createOpts)
+	if err != nil {
+		return fmt.Errorf("error creating Sagemaker endpoint configuration: %s", err)
+	}
+	d.SetId(name)
+
+	return resourceAwsSagemakerEndpointConfigurationRead(d, meta)
+}
+
+func resourceAwsSagemakerEndpointConfigurationRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).sagemakerconn
+
+	request := &sagemaker.DescribeEndpointConfigInput{
+		EndpointConfigName: aws.String(d.Id()),
+	}
+
+	endpointConfig, err := conn.DescribeEndpointConfig(request)
+	if err != nil {
+		if sagemakerErr, ok := err.(awserr.Error); ok && sagemakerErr.Code() == "ValidationException" {
+			log.Printf("[INFO] unable to find the sagemaker endpoint configuration resource and therefore it is removed from the state: %s", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("error reading Sagemaker endpoint configuration %s: %s", d.Id(), err)
+	}
+
+	if err := d.Set("arn", endpointConfig.EndpointConfigArn); err != nil {
+		return err
+	}
+	if err := d.Set("name", endpointConfig.EndpointConfigName); err != nil {
+		return err
+	}
+	if err := d.Set("production_variants", flattenProductionVariants(endpointConfig.ProductionVariants)); err != nil {
+		return err
+	}
+	if err := d.Set("kms_key_id", endpointConfig.KmsKeyId); err != nil {
+		return err
+	}
+
+	tagsOutput, err := conn.ListTags(&sagemaker.ListTagsInput{
+		ResourceArn: endpointConfig.EndpointConfigArn,
+	})
+	if err != nil {
+		return fmt.Errorf("error listing tags of Sagemaker endpoint configuration %s: %s", d.Id(), err)
+	}
+	if err := d.Set("tags", tagsToMapSagemaker(tagsOutput.Tags)); err != nil {
+		return err
+	}
+	return nil
+}
+
+func resourceAwsSagemakerEndpointConfigurationUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).sagemakerconn
+
+	d.Partial(true)
```

Comment:
Nit: Since this resource does not require partial Terraform state updates, we should remove the usage of `d.Partial()` and `d.SetPartial()`